src/components/Picker/BasePicker/index.js

```diff
@@ -1,32 +1,53 @@
 import React from 'react';
 import RNPickerSelect from 'react-native-picker-select';
+import _ from 'underscore';
 
 import styles from '../../../styles/styles';
 import * as basePickerPropTypes from './basePickerPropTypes';
 import basePickerStyles from './basePickerStyles';
 
-const BasePicker = props => (
-    <RNPickerSelect
-        onValueChange={props.onChange}
-        items={props.items}
-        style={props.size === 'normal' ? basePickerStyles(props.disabled, props.hasError, props.focused) : styles.pickerSmall}
-        useNativeAndroidPickerStyle={false}
-        placeholder={props.placeholder}
-        value={props.value}
-        Icon={() => props.icon(props.size)}
-        disabled={props.disabled}
-        fixAndroidTouchableBug
-        onOpen={props.onOpen}
-        onClose={props.onClose}
-        pickerProps={{
-            onFocus: props.onOpen,
-            onBlur: props.onClose,
-        }}
-    />
-);
+class BasePicker extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.state = {
+            selectedValue: this.props.value || this.props.defaultValue,
+        };
+
+        this.updateSelectedValueAndExecuteOnChange = this.updateSelectedValueAndExecuteOnChange.bind(this);
+    }
+
+    updateSelectedValueAndExecuteOnChange(value) {
+        this.props.onInputChange(value);
```

Comment:
See my comment above about `onChange` and `onInputChange`. We should rename `onChange` to `onInputChange` in all Picker usages.